kura/org.eclipse.kura.core/src/main/java/org/eclipse/kura/core/ssl/SslManagerServiceImpl.java

```diff
@@ -107,6 +109,14 @@ public void setKeystoreService(KeystoreService keystoreService, final Map<String
 
     public void unsetKeystoreService(KeystoreService keystoreService) {
         if (this.keystoreService == keystoreService) {
+
+            this.clearSslContexCache();
+
+            if (this.sslServiceListeners != null) {
+                // Notify listeners that service has been updated
+                this.sslServiceListeners.onConfigurationUpdated();
+            }
+
             this.keystoreService = null;
```

Comment:
I think the listeners should be called after setting the KeystoreService to `null`, otherwise the old keystore will be still be available when they are called.

Reply:
you are right